### ISSUE TITLE
[Accessibility] "Collapse panes" buttons should be accessible

### ISSUE THREAD

**Issue by yzen** [devtools-html/debugger.html#4073]
At the moment the buttons are not accessible to keyboard or screen reader users:

- [ ] Buttons should be reachable with the keyboard (e.g. should be focusable)
- [ ] Buttons should have hover/focus styling
- [ ] Buttons should actually be <BUTTON>s or at least have the ARIA button roles.
- [ ] Buttons should be actionable with the keyboard - Enter/Space should collapse/expand panes.

### Screenshots

![Screen Shot 2017-09-18 at 8.13.49 PM.png](https://shipusercontent.com/4e899cfb5b4a35bbe68a1d4dec33883f/Screen%20Shot%202017-09-18%20at%208.13.49%20PM.png)

**Comment by jasonLaster** [comment]
### Buttons should actually be buttons or at least have the ARIA button roles.

```diff
diff --git a/src/components/shared/Button/PaneToggle.js b/src/components/shared/Button/PaneToggle.js
index 9ce458a..b4a42bf 100644
--- a/src/components/shared/Button/PaneToggle.js
+++ b/src/components/shared/Button/PaneToggle.js
@@ -30,7 +30,7 @@ class PaneToggleButton extends Component {
       : L10N.getStr("collapsePanes");

     return (
-      <div
+      <button
         className={classnames(`toggle-button-${position}`, {
           collapsed,
           vertical: horizontal != null ? !horizontal : false
@@ -39,7 +39,7 @@ class PaneToggleButton extends Component {
         title={title}
       >
         <Svg name="togglePanes" />
-      </div>
+      </button>
     );
   }
 }
 ```
 
 #### listen for spacebar
 
 ```diff
 diff --git a/src/components/shared/Button/PaneToggle.js b/src/components/shared/Button/PaneToggle.js
index 9ce458a..eb1cb8a 100644
--- a/src/components/shared/Button/PaneToggle.js
+++ b/src/components/shared/Button/PaneToggle.js
@@ -15,6 +15,11 @@ type Props = {
 class PaneToggleButton extends Component {
   props: Props;

+  constructor(props) {
+    this.onKeyPress = this.onKeyPress.bind(this);
+    super(props);
+  }
+
   shouldComponentUpdate(nextProps: Props) {
     const { collapsed, horizontal } = this.props;

@@ -23,6 +28,8 @@ class PaneToggleButton extends Component {
     );
   }

+  onKeyPress(e) {}
+
   render() {
     const { position, collapsed, horizontal, handleClick } = this.props;
     const title = !collapsed
@@ -30,16 +37,17 @@ class PaneToggleButton extends Component {
       : L10N.getStr("collapsePanes");

     return (
-      <div
+      <button
         className={classnames(`toggle-button-${position}`, {
           collapsed,
           vertical: horizontal != null ? !horizontal : false
         })}
         onClick={() => handleClick(position, collapsed)}
         title={title}
+        onKeyPress={this.onKeyPress}
       >
         <Svg name="togglePanes" />
-      </div>
+      </button>
     );
   }
 }
```
 
#### Qs

> Buttons should be reachable with the keyboard 

should we do anything else if it is a button to be focusable? Is there an easy way to test this?

> Buttons should have hover/focus styling

What is a good default style for this kind of button w/ an svg

**Comment by yzen** [comment]
### A's
Nothing else needs to be done. Button will be in tab order, in fact, no need for special key press handling stuff. <BUTTON> will trigger "click" event when Enter/Space are pressed for free.

I think in for these sort of buttons, switching the drawing color to "focused" one (e.g. blue) would be best. Outline in this case could be avoided. For example, Firefox devtools toolbox tab that is selected changes icon color and text to blue.

You might also want to avoid and unset -moz-appearance for these buttons.

**Comment by Garbee** [comment]
Literally, just making these nodes `button` elements and fixing the main styling to be right resolves all of these issues. The styling of focus state can happen once button styling is figured out.
